### ISSUE TITLE
Feature Request: Implement OAuth#authenticate

### ISSUE THREAD

**Issue by moorage** [pelle/oauth-plugin#1]
Twitter implemented a great "sign-in with twitter" solution that utilizes OAuth. It would be great to provide this method in the OAuth controller!  Read about it here:

http://apiwiki.twitter.com/Sign-in-with-Twitter

**Comment by kerskine** [comment]
agreed - this would make my current app simplier in that it only goes back to the providers site/API
